src/flowchem/devices/knauer/knauer_autosampler.py

```diff
@@ -326,12 +326,27 @@ async def syringe_valve_position(self, port: str = None):
             raw_reply = await self._query(command_string) - 1
             return SwitchSyringeValveCommand.syringe_valve_positions(raw_reply).name
 
-    # tested
-    def injector_valve_position(self, port:str = None):
-        return self._set_get_value(SwitchInjectorValveCommand, port, SwitchInjectorValveCommand.allowed_position, get_actual=True)
-    
-    def needle_vertical_offset(self, offset: float = None):
-        return self._set_get_value(VerticalNeedleOffsetCommand, offset)
+    async def injector_valve_position(self, port: str = None):
+        return await self._set_get_value(SwitchInjectorValveCommand, port, SwitchInjectorValveCommand.allowed_position,
+                                         get_actual=True)
+
+    async def set_raw_position(self, identifier: str = None, position: str = None):
+        match identifier:
+            case "injection_valve":
+                return await self.injector_valve_position(port=position)
+            case "syringe_valve":
```

Comment:
basically, what you need is to inherit from the appropriate valve and then define the offset steps so the positions match